pylops/basicoperators/FirstDerivative.py

```diff
@@ -95,50 +114,50 @@ def __init__(
     def _matvec_forward(self, x):
         ncp = get_array_module(x)
         x = ncp.reshape(x, self.dims)
-        if self.dir > 0:  # need to bring the dim. to derive to first dim.
-            x = ncp.swapaxes(x, self.dir, 0)
+        if self.axis > 0:  # need to bring the dim. to derive to first dim.
```

Comment:
This could have been done before as it is an internal change, but since we are already working with moving all to axis=-1, I think it makes more sense to have this code pattern:

```
        if self.axis not in (-1, len(self.dims)-1):
            x = ncp.swapaxes(x, self.axis, -1)
        y = ncp.zeros(x.shape, self.dtype)
        y[..., :-1] = (x[..., 1:] - x[..., :-1]) / self.sampling
        if self.axis not in (-1, len(self.dims)-1):
            y = ncp.swapaxes(y, -1, self.axis)
```
as every operation should be faster (or at least not slower) in the last dimension. If you agree, the same needs changed in quite a few places :)

I can implement it but I just want to first check that you agree :)

Reply:
Actually, we can get rid of the `if` everywhere since `swapaxes` is extremely fast (less than a microsecond on my machine).  In case we both agree to just get rid of it, I can work on it but for another PR :)

Reply:
Agree, if it is such a minor overhead let's not bother checking and just do it :) another PR sounds good